src/target/riscv/riscv.c

```diff
@@ -3563,6 +3637,17 @@ static int register_set(struct reg *reg, uint8_t *buf)
 	memcpy(reg->value, buf, DIV_ROUND_UP(reg->size, 8));
 	reg->valid = gdb_regno_cacheable(reg->number, true);
 
+	if (reg->number == GDB_REGNO_TDATA1 ||
+			reg->number == GDB_REGNO_TDATA2) {
+		r->manual_hwbp_set = true;
```

Comment:
So, once set to true, manual_hwbp_set stays true for the duration of the current OpenOCD session, which makes sense.  If no direct accesses are made in the entire session, then the original logic is used for disabling/reenabling triggers (presumably that has less overhead?).

Reply:
That's right. In theory we could be smarter by tracking what tselect was, and what value was written to tdata, but that's a lot more complicated and I'm not sure how helpful it really is.